contracts/poll/RewardPoll.sol

```diff
@@ -33,6 +33,8 @@ contract RewardPoll is BasePoll, Roles {
         uint256 _duration,
         address _poolAddress,
         address _tokenAddress
+    // warning: the length of the poll is dependent on the time the block is mined.
```

Comment:
Only if you have logic outside of the contract depending on that variable right? I think for now we can live with that since the only risk I see is users having to wait a little longer before being able to finalize a poll or being just to late to cast a vote. 

What could be an alternative? Passing a timestamp in the method?

Reply:
Yes I think so too. But if users keep an internal state to check when the poll is ending (e.g. now + duration) it will differ with the smart contract. As the time to mine the block is added. `timestamp_end` will be sufficient to tackle this problem.

Reply:
Only a problem if system are keeping this internal state of course.